### ISSUE TITLE
Add MkDocs support with mkdocs.yml linting rules

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#354]
## Summary

Add support for linting MkDocs projects, starting with `mkdocs.yml` configuration validation and eventually MkDocs-specific markdown extensions.

## Motivation

MkDocs (especially with Material theme) is a popular alternative to mdBook for documentation sites. While the markdown content can already be linted, there's no good tooling for:
- Validating `mkdocs.yml` best practices (beyond schema validation)
- Checking MkDocs-specific markdown syntax (admonitions, tabs, etc.)
- Cross-file validation (nav entries exist, orphaned files, etc.)

Existing tools:
- Schema validation via Material's schema.json (catches type errors only)
- Generic YAML linters like yamllint (no MkDocs awareness)
- markdownlint (struggles with MkDocs admonition syntax)

### Phase 1: mkdocs.yml Configuration Rules

| Rule | Name | Description | Severity |
|------|------|-------------|----------|
| MKDOCS001 | site-name-required | `site_name` must be defined | Error |
| MKDOCS002 | site-url-recommended | `site_url` should be set for production | Warning |
| MKDOCS003 | nav-files-exist | All nav entries must point to existing files | Error |
| MKDOCS004 | theme-valid | Theme name should be valid (mkdocs, material, readthedocs) | Warning |
| MKDOCS005 | orphan-files | Warn about markdown files not in nav | Info |
| MKDOCS006 | nav-depth | Warn about deeply nested nav (>4 levels) | Warning |

### Phase 2: MkDocs-Aware Markdown Rules (Future)

| Rule | Name | Description |
|------|------|-------------|
| MKDOCS010 | admonition-syntax | Validate admonition syntax (`!!! note`, etc.) |
| MKDOCS011 | tabs-syntax | Validate content tabs syntax |
| MKDOCS012 | snippet-paths | Validate snippet file paths exist |
| MKDOCS013 | icon-references | Validate icon references (`:material-xxx:`) |

### New Types

```rust
// ConfigRule trait for YAML-based rules
pub trait ConfigRule: Send + Sync {
    fn id(&self) -> &'static str;
    fn name(&self) -> &'static str;
    fn description(&self) -> &'static str;
    fn metadata(&self) -> RuleMetadata;
    fn check_config(&self, config: &MkDocsConfig, raw: &str) -> Result<Vec<Violation>>;
}

// Adapter to make ConfigRule implement Rule
pub struct ConfigRuleAdapter<T: ConfigRule>;
```

### File Structure

```
crates/mdbook-lint-rulesets/src/
  mkdocs/
    mod.rs              # MkDocsRuleProvider, ConfigRule trait
    config.rs           # MkDocsConfig serde model
    mkdocs001.rs        # site-name-required
    mkdocs002.rs        # site-url-recommended
    ...
```

### CLI Integration

```bash
# New subcommand
mdbook-lint mkdocs [path]

# Examples
mdbook-lint mkdocs .                    # Lint mkdocs.yml in current dir
mdbook-lint mkdocs ./mkdocs.yml         # Lint specific file
mdbook-lint mkdocs ./my-docs-project    # Lint project directory
```

### Feature Flag

```toml
[features]
mkdocs = ["dep:yaml-rust2"]  # For accurate line number tracking
```

### YAML Parsing with Position Tracking

Use `yaml-rust2` instead of `serde_yaml` for accurate line/column numbers in violations. This requires manual traversal but gives precise error locations.

### New RuleCategory

Add `RuleCategory::MkDocs` to match existing `RuleCategory::MdBook`.

### Phase 1 (MVP)
- [ ] Add `mkdocs` feature flag to rulesets crate
- [ ] Add `yaml-rust2` dependency
- [ ] Create `MkDocsConfig` model with position tracking
- [ ] Implement `ConfigRule` trait and adapter
- [ ] Add `RuleCategory::MkDocs`
- [ ] Implement MKDOCS001 (site-name-required)
- [ ] Implement MKDOCS002 (site-url-recommended)
- [ ] Add `mkdocs` subcommand to CLI
- [ ] Add tests
- [ ] Document in rules-reference.md

### Phase 2 (Enhanced)
- [ ] MKDOCS003: nav-files-exist (requires filesystem access)
- [ ] MKDOCS004: theme-valid
- [ ] MKDOCS005: orphan-files
- [ ] MKDOCS006: nav-depth

### Phase 3 (Markdown Extensions)
- [ ] Research Material for MkDocs extension syntax
- [ ] MKDOCS010-013: Extension-specific rules

## References

- [MkDocs Configuration](https://www.mkdocs.org/user-guide/configuration/)
- [Material for MkDocs](https://squidfunk.github.io/mkdocs-material/)
- [Material schema.json](https://squidfunk.github.io/mkdocs-material/schema.json)
- [yaml-rust2](https://crates.io/crates/yaml-rust2)

**Comment by mschoettle** [comment]
For existing tools, there is also `rumdl` with the `mkdocs` flavour: https://github.com/rvben/rumdl/blob/8eeed34c6f8f3bacc43f59a6a34f3c15dd594b20/docs/flavors/mkdocs.md#L4

And there is also `mkdocs-material-linter` which is a `markdownlint-cli2` plugin that provides additional rules: https://github.com/mensfeld/mkdocs-material-linter

**Comment by joshrotenberg** [comment]
Closing as out of scope. This project is specifically an mdBook linter -- adding MkDocs support would be a significant scope expansion that isn't planned. If MkDocs linting is needed, existing tools like rumdl (mentioned in the comments) are better suited.
